pkg/processor/procbuiltin/unwrap.go

```diff
@@ -0,0 +1,310 @@
+		if err != nil {
+			return record.Record{}, cerrors.Errorf("failed to unmarshal raw data as JSON: %w", unwrapPayloadProcType, err)
+		}
+	case record.StructuredData:
+		structData = d
+	default:
+		return record.Record{}, cerrors.Errorf("unexpected data type %T", unwrapPayloadProcType, data)
+	}
+
+	// get payload
+	var structPayload record.StructuredData
+	payload := structData["payload"]
+	if p, ok := payload.(map[string]interface{}); ok {
+		structPayload = p
+	} else {
+		return record.Record{}, cerrors.Errorf("payload doesn't contain a record")
+	}
+
+	return record.Record{
+		Key:      rec.Key,
+		Position: rec.Position,
+		Metadata: nil,
+		Payload: record.Change{
+			Before: nil,
+			After:  structPayload,
+		},
+		Operation: record.OperationSnapshot,
+	}, nil
+}
+
+/*
+	 example of debezium record:
+		`{
+		 "payload": {
+		   "after": {
+		     "description": "desc",
+		     "id": 20
+		   },
+		   "before": null,
+		   "op": "c",
+		   "source": {
+		     "opencdc.readAt": "1674061777225877000",
+		     "opencdc.version": "v1",
+		   },
+		   "transaction": null,
+		   "ts_ms": 1674061777225
+		 },
+		 "schema": {} // will be ignored
+		}`
+*/
+type DebeziumUnWrapper struct{}
+
+const (
+	DebeziumOpCreate = "c"
+	DebeziumOpUpdate = "u"
+	DebeziumOpDelete = "d"
+	DebeziumOpRead   = "r" // snapshot
+
+	DebeziumFieldBefore  = "before"
+	DebeziumFieldAfter   = "after"
+	DebeziumFieldSource  = "source"
+	DebeziumFieldOp      = "op"
+	DebeziumFieldPayload = "payload"
+)
+
+func (d *DebeziumUnWrapper) Unwrap(rec record.Record) (record.Record, error) {
+	data := rec.Payload.After
+	var structData record.StructuredData
+	switch d := data.(type) {
+	case record.RawData:
+		// todo: take this section out, after platform team support ordering processors
+		// unmarshal raw data to structured
+		err := json.Unmarshal(data.Bytes(), &structData)
+		if err != nil {
+			return record.Record{}, cerrors.Errorf("failed to unmarshal raw data as JSON: %w", unwrapPayloadProcType, err)
+		}
+	case record.StructuredData:
+		structData = d
+	default:
+		return record.Record{}, cerrors.Errorf("unexpected data type %T", unwrapPayloadProcType, data)
+	}
+
+	// get payload
+	var structPayload record.StructuredData
+	payload := structData["payload"] // the payload has the debezium record
+	if p, ok := payload.(map[string]interface{}); ok {
+		structPayload = p
+	} else {
+		return record.Record{}, cerrors.Errorf("payload doesn't contain a record")
+	}
+
+	// check fields under payload
+	err := d.validateRecord(structPayload)
+	if err != nil {
+		return record.Record{}, err
+	}
+
+	var before record.StructuredData
+	if b, ok := structPayload[DebeziumFieldBefore].(map[string]any); ok {
+		before = b
+	}
+
+	var after record.StructuredData
+	if a, ok := structPayload[DebeziumFieldAfter].(map[string]any); ok {
+		after = a
+	}
+
+	var op string
+	if str, ok := structPayload[DebeziumFieldOp].(string); ok {
+		op = str
+	} else {
```

Comment:
I suggest a simplification.
```suggestion
	op, ok := structPayload[DebeziumFieldOp].(string)
	if !ok {
```